src/DFMM.sol

```diff
@@ -108,24 +110,21 @@ contract DFMM is IDFMM {
 
         for (uint256 i = 0; i < tokensLength; i++) {
             address token = params.tokens[i];
-
-            for (uint256 j = 0; j < tokensLength; j++) {
-                if (i != j && token == params.tokens[j]) {
-                    revert InvalidDuplicateTokens();
-                }
-            }
-        }
-
-        for (uint256 i = 0; i < tokensLength; i++) {
             uint256 decimals = ERC20(params.tokens[i]).decimals();
```

Comment:
You can use `token` here instead of `params.tokens[i]` again:

```diff
  address token = params.tokens[i];
- uint256 decimals = ERC20(params.tokens[i]).decimals();
- uint256 decimals = ERC20(token ).decimals();
```

Reply:
Fixed here https://github.com/primitivefinance/DFMM/pull/119/commits/e1d1a4dd0a73dfbb21512b820484664aac8ec991.